### PR TITLE
[TT-974] Chainlink-evm workflow dispatch updates

### PR DESCRIPTION
**Pull request by tateexon**
Fix up the ci-core and integration-tests workflows to work with the GATI workflow dispatch from chainlink-evm

### CODE REVIEW

File: .github/workflows/ci-core.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 name: CI Core
+run-name: CI Core ${{ inputs.distinct_run_name && inputs.distinct_run_name || '' }}
```
Comment (tateexon) [inline .github/workflows/ci-core.yml:2]: The distinct_run_name usage is required here to properly find the running workflow after it has kicked off. Without this there is a chance we grab the wrong one when the repo is running workflows for several PRs around the same time. Github doesn't provide a built in way to get this in the api when doing a workflow_dispatch so we have to work around that.

File: .github/workflows/ci-core.yml [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ inputs.distinct_run_name }}
```
Comment (tateexon) [inline .github/workflows/ci-core.yml:5]: Added this here so if chainlink-evm pushes two PRs that run these tests around the same time in the future they don't collide with the last one in winning.

File: .github/workflows/ci-core.yml [hunk 3]
```diff
@@ -102,6 +114,10 @@ jobs:
       - name: Download Go vendor packages
         if: ${{ needs.filter.outputs.changes == 'true' }}
         run: go mod download
+      - name: Replace chainlink-evm deps
+        if: ${{ inputs.evm-ref != ''}}
```
Comment (kalverra) [inline .github/workflows/ci-core.yml:118]: Want to also check for `needs.filter.outputs.changes == 'true'` ?

**Review by kalverra — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-core.yml (hunk 3 above)
Comment (kalverra) [reply .github/workflows/ci-core.yml:118]: Speaking of, would the `needs` job that checks for changes inadvertently kill this workflow when doing the dispatch trigger?

Comment (tateexon) [reply .github/workflows/ci-core.yml:118]: @kalverra I think I started working on this before the filtering was added so I will need to look again. As for the filter itself I don't know if it is setup correctly or not. I haven't used it in a negating way before, will it currently only run if there are changes only outside of the integration tests directory? Will it still run if there are changes in both?

Comment (tateexon) [reply .github/workflows/ci-core.yml]: @kalverra The lack of the needs for the filter didn't seem to matter and I am not sure why exactly. I put the same input to force the filter to true for workflow dispatch events that I had in the integration-tests workflow so at least the logic makes sense for why it runs now.

**Review by kalverra — APPROVED** [review]

**Pull request merged**